### PR TITLE
improvement: add deletion script

### PR DESCRIPTION
**Pull request by alban23**
Add deletion script so that the stack can be easily cleaned up when not
needed anymore. Further, the script adjusts the region automatically, so only
one script is needed for Ireland and Virginia. Also, a name tag is added to
the resources created by the script. Moreover, useful commands and resource
ids are saved into a text file.

Also relates to https://github.com/fastai/courses/issues/6.

### CODE REVIEW

**Comment by alban23** [comment]
Please review carefully and thoroughly, I have tried it only on my mac os laptop.

File: setup/setup_p2.sh [hunk 1]
```diff
@@ -1,39 +1,117 @@
...
 aws ec2 wait instance-running --instance-ids $instanceId
 sleep 10 # wait for ssh service to start running too
 export assocId=`aws ec2 associate-address --instance-id $instanceId --allocation-id $allocAddr --query 'AssociationId' --output text`
 export instanceUrl=`aws ec2 describe-instances --instance-ids $instanceId --query 'Reservations[0].Instances[0].PublicDnsName' --output text`
-echo securityGroupId=$securityGroupId
-echo subnetId=$subnetId
-echo instanceId=$instanceId
-echo instanceUrl=$instanceUrl
-echo Connect: ssh -i ~/.ssh/aws-key.pem ubuntu@$instanceUrl
+export ebsVolume=`aws ec2 describe-instance-attribute --instance-id $instanceId --attribute  blockDeviceMapping  --query BlockDeviceMappings[0].Ebs.VolumeId --output text`
```
Comment (alexanderkjeldaas) [inline setup/setup_p2.sh]: what are you using the EBS volume for?

File: setup/setup_p2.sh [hunk 2]
```diff
@@ -1,39 +1,117 @@
...
+echo \# Connect to your instance: > $name-commands.txt # overwrite existing file
+echo ssh -i ~/.ssh/aws-key-$name.pem ubuntu@$instanceUrl >> $name-commands.txt
+echo \# Stop your instance: : >> $name-commands.txt 
+echo aws ec2 stop-instances --instance-ids $instanceId  >> $name-commands.txt
+echo \# Start your instance: >> $name-commands.txt 
+echo aws ec2 start-instances --instance-ids $instanceId  >> $name-commands.txt
+echo Reboot your instance: >> $name-commands.txt 
+echo aws ec2 reboot-instances --instance-ids $instanceId  >> $name-commands.txt
+echo "" 
+# export vars to be sure
```
Comment (alexanderkjeldaas) [inline setup/setup_p2.sh]: Define a version for this setup, as this script is already incompatible with what people have running, and further changes will further make it incompatible.

At what point should this be done in cloudformation?

**Comment by alban23** [comment]
Thank you very much for your review. I added my script now as a new file instead of overwriting existing scripts. I used the EBS id to check if the volume gets deleted automatically, now I commented it out. 

Yes, CloudFormation is definitely the way to go with this and what I would normally use (that is why this script is currently far from perfect, I just wanted to play around with the CLI in timeboxed way). It would make the script a 5 liner and the definitions could be nicely saved in a yaml file. Should we create an issue for that?

**Comment by alexanderkjeldaas** [comment]
fyi, I have nothing to do with this course, I'm just reviewing because I want some improvements to go in.  I don't have any special privileges in the project.

**Comment by racheltho** [comment]
Thanks for doing this!  I added and committed these changes from master since I also needed to rename the files, delete the old setups, include Oregon, and add the AMI IDs for t2 instances.

**Pull request closed without merging**